tests/test_lambda.py

```diff
@@ -41,7 +45,11 @@ def lambda_handler(event, context):
     return _process_lambda(lambda_src)
 
 
-async def test_run_lambda(iam_client, lambda_client, aws_lambda_zip):
```

Comment:
https://github.com/getmoto/moto/pull/8603 should take care of the root cause of our failing tests.

Reply:
ok lemme re-lock, I'll keep the env var in case we need it later

Reply:
ah 29 is not out yet, added cmt to remove once it's released

Reply:
I believe moto dropped support for Python 3.8 in the same release they shipped that fix. Since we still support Python 3.8, we might be blocked from upgrading for the time being.

Reply:
> ok lemme re-lock, I'll keep the env var in case we need it later

Could you at least do that work in a separate PR? I don't think this is related to the original work.

Reply:
py3.8 is no longer supported, we should drop it: https://devguide.python.org/versions/

Reply:
#1302 bumps dev dependencies, including moto. That takes care of failing tests, except for Python 3.8 on arm64.

Reply:
we're kinda unique requirements as we require more things than botocore which could cause us to be a tighter version set, however meh, might as well keep it for now

Reply:
The underlying issue is not specific to GitHub CI: If you run tests locally with Python 3.8 on Apple silicon, you'd also be affected, i.e. 50% of runs would fail.

If you really want to address this, we should look at sys.version and something like https://stackoverflow.com/questions/71548156/how-to-get-architecture-of-running-python-interpreter-on-macos-with-apple-silico. 

Since Python 3.8 support will be dropped rather sooner than later, I'd say ignore it a tiny bit longer.

The light-weight fix would be to ignore platforms, and simply skip that test on Python 3.8 (via pytest decorator using sys.version).

In any case, I strongly suggest to open a separate PR.

Reply:
I actually prefer it to fail when run locally as this needs to be resolved.  I only care during prod builds so it doesn't interrupt PRs from getting merged.  Given I needed it to get this PR to pass I suggest keeping here

Reply:
> I actually prefer it to fail when run locally as this needs to be resolved.
Well it won't, unless moto backports my PR and does another 4.0.x release with Python 3.8 support. But ok.

> I only care during prod builds so it doesn't interrupt PRs from getting merged. Given I needed it to get this PR to pass I suggest keeping here
That is incorrect: We treat arm jobs as experimental, i.e. they DO NOT BLOCK CI.

BTW, I am currently prepping a PR that "graduate" arm from experimental stage, except for Python 3.8. That way, if something else fails, it would block CI going forward.

Reply:
There we go: #1303

Reply:
> BTW, I am currently prepping a PR that "graduate" arm from experimental stage, except for Python 3.8. That way, if something else fails, it would block CI going forward.

hmm well they were showing as failed here so if it didn't block CI was still confusing.  But better just not run it like you did

Reply:
reverted per your PR

Reply:
> hmm well they were showing as failed here so if it didn't block CI was still confusing.

I agree, GitHub UI is confusing in that regard.